python/lsst/ap/association/diaPipe.py

```diff
@@ -405,8 +408,8 @@ def run(self,
         ----------
         diaSourceTable : `pandas.DataFrame`
             Newly detected DiaSources.
-        solarSystemObjectTable : `pandas.DataFrame`
-            Preloaded Solar System objects expected to be visible in the image.
+        legacySolarSystemTable : `pandas.DataFrame`
+            Not used
```

Comment:
The idea is to maintain the interface's backwards compatibility -- removing the argument in the place of `legacySolarSystemTable` would break every place where `diaPipe.run` is currently being called. I do not believe there is a plan for phasing this out.

Reply:
Except that what you have here is not actually backward-compatible -- a hypothetical user who tries to call `run` with the old signature, expecting to get solar system associations, will silently get nothing. If you _do_ want backward compatibility, you'd need to convert the old-style input into the new-style input.

Phasing out could be handled [by raising a `FutureWarning`](https://developer.lsst.io/stack/deprecating-interfaces.html) if the argument isn't `None` (there's no specific procedure for deprecating parameters, but see that page for recommended text) and filing a Jira issue to remove it later.

All that said, though, I doubt that `run` is called anywhere except from `runQuantum` or unit tests, so I think you could get away with unceremoniously deleting it.

Reply:
After discussing with @Gerenjie and @isullivan separately, we've agreed to move forward as follows:

- `legacySolarSystemTable` will be used in place of `solarSystemObjectTable` when the latter is undefined. That should be enough to give backwards-compatibility whether `run` is called with positional args or keywords, and is the only change that will be made right now.
- _After_ merging this PR, we'll file an RFC to deprecate and remove `legacySolarSystemTable` [per the usual process](https://developer.lsst.io/stack/deprecating-interfaces.html). I doubt there will be any pushback.